### PR TITLE
Eternal season temperature fix

### PR DESCRIPTION
**Pull request by 1942rob**
Fixed bug where the yearly temperature-cycle happened even when eternal season was enabled, e.g. the first summer would get summer temperatures, then the next summer would get spring temperatures and so on.

### CODE REVIEW

File: src/calendar.cpp [hunk 1]
```diff
@@ -398,6 +398,14 @@ int calendar::season_length()
     return cached_season_length;
 }
 
+int calendar::turn_of_year() const {
+	return (season_turns() * season) + (turn_number % season_turns());
```
Comment (BevapDin) [inline src/calendar.cpp]: Why did you change the meaning of this function?

Comment (1942rob) [inline src/calendar.cpp]: It still has the same functionality as before, but if eternal season is enabled it will only return the turn of year corresponding to that season.
weather_generator::get_weather(...)  can then simulate temperatures for 4 equal seasons in a year.

Comment (BevapDin) [inline src/calendar.cpp]: The old function contains essentially this logic:
```C++
int turn_of_year() const { 
    return turn_number % DAYS( season_length() * 4 );
}
```

The new function has this:
```C++
int calendar::turn_of_year() const { 
    return DAYS( season_length() ) * season + turn_number % DAYS( season_length() ); 
}
```

According to `calendar::sync`, `season` has either the value `initial_season` (for eternal season) or `turn_number / DAYS( season_length() ) % 4)`.

If the current time is 2 days into the second season (`turn_number = DAYS( season_length() ) + DAYS( 2 )`, `season` be `1` (non-eternal, see `sync`). The old function would return `turn_number`, the new function returns `turn_number + DAYS( season_length() )`.

The function would never return anything `>= DAYS( season_length() * 4 )`, now it can.

Comment (1942rob) [inline src/calendar.cpp]: uhm, the new function would return DAYS( season_length() + turns_into_current_season which equals turn_number

Comment (BevapDin) [inline src/calendar.cpp]: You're right, I must have missed something.

You should replace the tabs with 4 spaces (see doc/CODE_STYLE.txt) and apply the style to changed code (you can use `astyle` for this).

Comment (1942rob) [inline src/calendar.cpp]: no prob, and will do

**Comment by wdurocher33** [comment]
This issue is still present with build 7710. Version 0.C -30680 - gb8ef478

Some settings I have set:
Initial season:  Summer
Season length: 14
Eternal season: True

Note that the display does say Summer, day 11 xx:xx:xx
